wicket-jquery-ui/src/main/java/com/googlecode/wicket/jquery/ui/form/autocomplete/AutoCompleteTextField.java

```diff
@@ -323,9 +340,42 @@ protected CharSequence getChoiceCallbackUrl()
 
 				return super.$();
 			}
+
+			@Override
+			public void onConfigure(Component component) {
+				super.onConfigure(component);
+				if (preventSelectWhileQueryIsRunning) {
+					// we define special functions in for handling select and fetch data
+					this.setOption("select", "function(request, response) {\n" + getVarName() + ".select(request, response); \n}");
```

Comment:
I am not familiar with WJQUI and jQueryUI JS API but the above line looks suspicious to me. The JS code does not look like it prevents the selection. How does this work exactly ?

Reply:
This just delegates into function defined in prototype.